src/build.py

```diff
@@ -288,6 +288,15 @@ def IsBuildbot():
   flags = fcntl(fd, F_GETFL)
   fcntl(fd, F_SETFL, flags & ~os.O_NONBLOCK)
 
+  # Use gclient-managed hermetic toolchain on mac LUCI bots.
+  # TODO: Convert bot build to recipes, and this should no longer be necessary.
+  # Force gclient to install the hermetic mac toolchain
+  os.environ['FORCE_MAC_TOOLCHAIN'] = 1
+  # The V8 build_overrides/build.gni specifically forces use of the system
+  # Xcode, so override the override.
+  os.environ['DEVELOPER_DIR'] = os.path.join(V8_SRC_DIR,
+                                             'build', 'mac_files', 'Xcode.app')
```

Comment:
This feels like it should be part of V8 build only, no?   Or does it effect other builds too?

Reply:
The intent was to make it affect all the builds, since in general it seems best to use the same toolchain everywhere. I still don't understand why the non-V8 builds are working when the V8 build is not, though (maybe the failing check is overly-specific? Don't know).  Eventually I think we want to move this bot fully over to LUCI-style recipes (and my understanding is that LUCI bots have a different way of managing the toolchain than I'm using here). So this is supposed to be a temporary fix.

In any case I also found that none of this is actually running anyway because LUCI builders don't set `BUILDBOT_BUILDNUMBER` along with the other environment vars. So I also need to figure out how to get the LUCI build number if we want to archive any builds.